src/content/snapshot/troubleshooting-snapshots.md

```diff
@@ -232,6 +232,28 @@ Yes, check out the Chromatic's [modes feature](/docs/modes) is simplifies the pr
 
 </details>
 
+<details>
+
+  <summary>Why is my mocked data not being snapshotted correctly?</summary>
+
+Chromatic snapshots sometimes show the initial or intermediate loading state of the UI, instead of the final state with the mocked data. This can lead to visual tests failing, even though the Storybook renders correctly locally. To debug this issue, follow these steps:
+
+1.  Ensure that MSW (Mock Service Worker) is correctly initialized in your Storybook configuration. Follow the instructions in the MSW documentation: [here](https://github.com/mswjs/msw-storybook-addon?tab=readme-ov-file#configure-the-addon).
+
+2.  Confirm that the versions of `msw`, `msw-storybook-addon`, or any other community add-ons you are using are not outdated.
+
+3.  Run `npm run build-storybook` and `npx http-server storybook-static -o` locally to check for console output and address any MSW warnings or errors. Even if warnings pass locally, they may not work on the Chromatic side.
+
+4.  Pay close attention to how you define MSW handlers, especially for API requests with query parameters. MSW recommends matching only the path in the handler URL and accessing query parameters inside the resolver function using `req.url.searchParams.get()`.
+
+5.  For Interaction tests, use the Storybook Test Runner to gain more insight into mock behavior. The Test Runner provides CI logs as pre-render and post-render information. Learn more [here](https://storybook.js.org/docs/writing-tests/test-runner#test-hook-api).
```

Comment:
@cgbl-90 This is great! I have one note. The Storybook team has somewhat deprecated the Storybook test-runner in favor of the [Storybook Test addon](https://storybook.js.org/docs/writing-tests/test-addon). The Test addon only works for Vite-based projects, but it is the recommended approach for getting more debugging insight into interaction tests.

I would keep what you have, as it is still relevant for Webpack projects, but I would also add something about the Storybook Test Addon.